### ISSUE TITLE
Promtail not working in dry-run mode

### ISSUE THREAD

**Issue by aronneagu** [grafana/loki#1900]
**Describe the bug**
Running promtail with dry-run arguments expects a client to be configured

**To Reproduce**
1. Installed promtail version 1.4.0
2. Ran `cat eventrouter.log | ./promtail -config.file promtail.yaml -dry-run` 

**Expected behavior**
Expected to see the log line how it will be passed to loki

**Environment:**
```
cat eventrouter.log
{"log":"I0406 15:50:11.766875       1 glogsink.go:42] {\"verb\":\"ADDED\",\"event\":{\"metadata\":{\"name\":\"minikube.16034541d120b660\",\"namespace\":\"default\",\"selfLink\":\"/api/v1/namespaces/default/events/minikube.16034541d120b660\",\"uid\":\"1d128c8b-fb24-4919-9fb1-789057bc0b77\",\"resourceVersion\":\"10931185\",\"creationTimestamp\":\"2020-04-06T15:40:42Z\"},\"involvedObject\":{\"kind\":\"Node\",\"name\":\"minikube\",\"uid\":\"322204d2-2ed6-44e3-b6c6-9e036633abbb\"},\"reason\":\"RegisteredNode\",\"message\":\"Node minikube event: Registered Node minikube in Controller\",\"source\":{\"component\":\"node-controller\"},\"firstTimestamp\":\"2020-04-06T15:40:42Z\",\"lastTimestamp\":\"2020-04-06T15:40:42Z\",\"count\":1,\"type\":\"Normal\",\"eventTime\":null,\"reportingComponent\":\"\",\"reportingInstance\":\"\"}}\n","stream":"stderr","time":"2020-04-06T15:50:11.767013811Z"}
```

**Screenshots, Promtail config, or terminal output**
```
cat event.log |./promtail -config.file promtail.yaml -dry-run
level=error ts=2020-04-06T16:17:13.377462255Z caller=main.go:63 msg="error creating promtail" error="at least one client config should be provided"
cat promtail.yaml
scrape_configs:
- job_name: test-job
  pipeline_stages:
    - docker: {}
```

I took the instructions from the releases page, for v 1.4.0, but maybe I am missing something. I would have expected promtail to just parse the input and output  what would be fed to loki, but without actually configuring any loki clients

**Comment by cyriltovena** [comment]
You need at least an url for dry-run

try 

```
cat event.log | ./promtail --config.file=promtail.yaml --dry-run --client.url=http://foo/
```

Are you using linux ? if so careful with the issue below.

https://github.com/grafana/loki/issues/1892

I think @slim-bean will release a 1.4.1

**Comment by aronneagu** [comment]
Yes, I am using Linux. I have tried adding in a `-client.url localhost:8080` (with netcat listening on that port) but no luck. I will try on Windows and report back. 
By the way, do the cli arguments need one 
dash or two dashes? `promtail - h` shows args with one dash

**Comment by cyriltovena** [comment]
You don’t need netcat, the url is just there to show you to which server it
will send, but won’t send anything.

Single dash should work, it won’t work on windows either we need to give
you a new binary there’s a bug in 1.4.0 which will be fixed in 1.4.1 today
or tomorrow.

**Comment by aronneagu** [comment]
Awesome, looking forward for the new release. Thanks for the lightning fast replies

You are doing an amazing work with loki and promtail.
